code/__HELPERS/jatum.dm

```diff
@@ -80,15 +80,13 @@
 		// Serialize all lists as dicts, list("a") and list("a" = null) can't be differentiated in DM
 		var/list_contents = list()
 		for(var/key in value)
-			var/got_l_value
 			var/l_value
 			try
 				l_value = value[key]
-				got_l_value = TRUE
 			catch
 				// Expected, indicates a flat list
 
-			if(got_l_value)
```

Comment:
Since the previous method was bugged, we actually need to somehow test if we have a flat or keyed list before we start building `list_contents` here.

Reply:
l_value is null in case of a flat list